src/jit/codegenxarch.cpp

```diff
@@ -520,6 +520,38 @@ void CodeGen::genCodeForNegNot(GenTree* tree)
     genProduceReg(tree);
 }
 
+//------------------------------------------------------------------------
+// genCodeForBswap: Produce code for a GT_BSWAP node.
+//
+// Arguments:
+//    tree - the node
+//
+void CodeGen::genCodeForBswap(GenTree* tree)
+{
+    // TODO: If we're swapping immediately after a read from memory or immediately before
```

Comment:
Presumably that's why you didn't just use `genCodeForNegNot`?

Reply:
i.e. because the handling of a memory operand (containment) is different in this case because `MOVBE` has both a source memory location and a register destination (unlike neg and not which take a single operand that's modified).

Reply:
@mikedn Per your other comment, I lifted knowledge of the `INS_bswap` instruction directly to this method, so now I think the logic is different enough that it's worth keeping negnot and bswap split. Is there still a preference to merge the two?

@CarolEidt This is useful feedback, thanks! I'm going to open a different issue for `movbe` support since I think it's complex enough to warrant a different PR. (It requires a cpuid capability check so it affects AoT scenarios, and I haven't yet created a C++ benchmark app demonstrating that it's a worthwhile optimization.)